src/pickley/__init__.py

```diff
@@ -7,17 +7,27 @@
 from datetime import datetime
 
 import runez
-from runez.pyenv import PypiStd, PythonDepot, Version
+from runez.pyenv import ArtifactInfo, PypiStd, PythonDepot, Version
 
-__version__ = "4.1.4"
+from pickley.bstrap import DOT_META, http_get, PICKLEY
+
+__version__ = "4.2.0rc1"
 LOG = logging.getLogger(__name__)
-PICKLEY = "pickley"
-DOT_META = ".pk"
 K_CLI = {"delivery", "index", "python"}
 K_DIRECTIVES = {"include"}
 K_GROUPS = {"bundle", "pinned"}
-K_LEAVES = {"facultative", "install_timeout", "preferred_pythons", "python_installations", "pyenv", "version", "version_check_delay"}
+K_LEAVES = {
+    "facultative",
+    "install_timeout",
+    "package_manager",
+    "preferred_pythons",
+    "python_installations",
+    "pyenv",
+    "version",
+    "version_check_delay",
+}
 PLATFORM = platform.system().lower()
+RX_HREF = re.compile(r'href=".+/([^/#]+\.(tar\.gz|whl))#', re.IGNORECASE)
```

Comment:
This is a non-exhaustive list, in particular .tar.bz2 and .zip are fairly common.  See https://github.com/jwodder/pypi-simple/blob/ba1fec22b4af2294b2cae152342d4bdc3ccd9f94/src/pypi_simple/filenames.py#L10

Reply:
Would these ever come up as sole artifacts for a CLI? The idea here is to simply parse the `/simple` html for versions (nothing besides the available versions is relevant to pickley)